functorch/experimental/_cond.py

```diff
@@ -149,6 +158,100 @@ def cond_python_dispatcher(*args):
     return cond(*args)
 
 
+def _has_potential_branch_input_mutation(branch, fake_inputs):
+    """
+    Dispatch-trace the branch with fake inputs and check if
+    producing graph has mutable op on the input or its' alias. This is
+    bit restrictive as the branch must be traceable.
```

Comment:
Nit: the comment is a bit misleading, right? My reading of this function is that it detects if the input gets mutated in-place, but it won't detect if the `branch` does x[0].sin_(), so this function is not a general function to detect if the input's alias gets mutated.

It all works out in the cond implementation below because we end up passing a functionalized function as `branch`, and the functionalized function has a clause somewhere that replays mutations to the input.